lib/elixir/lib/integer.ex

```diff
@@ -11,9 +11,35 @@ defmodule Integer do
   Returns `true` if `n` is an odd number, otherwise `false`.
 
   Allowed in guard clauses.
+
+  ## Examples
+
+      iex> Integer.is_odd(3)
+      true
+
+      iex> Integer.is_odd(4)
+      false
   """
+  @spec is_odd(integer) :: boolean
   defmacro is_odd(n) do
-    quote do: (unquote(n) &&& 1) == 1
+    case Macro.Env.in_guard?(__CALLER__) do
+      true ->
+        quote do: (unquote(n) &&& 1) == 1
+
+      false ->
+        quote bind_quoted: [n: n] do
+          number = fn -> n end
+          case is_integer(number.()) do
```

Comment:
this is the only work around i found to avoid getting a warning, every time i call this function:

> test/elixir/integer_test.exs:18: warning: this clause cannot match because a previous clause at line 18 always matches